### DISCUSSION TITLE
Localised annotations not detected by Cucumber VSCode extension

### DISCUSSION THREAD

**Discussion by Tavo521** [cucumber/common#2186] [Visual Studio Code] [answered]
I have problems with the extension, I have some scenarios written in Spanish that work correctly in the Intellij IDE, but in VSC it shows the following message:

![image](https://github.com/user-attachments/assets/a04d6310-fb14-4621-8d5e-433d6a26b014)

**But the scenario is defined** 
![image](https://github.com/user-attachments/assets/53998c33-41ed-40b4-a1df-6599ea56883a)

I have tried restarting VSC and also reinstalling the cucumber extension but it still doesn't work.

**this is my settings.json file** 
![image](https://github.com/user-attachments/assets/a112f7fc-f186-4f65-9295-9ac80f3da9ea)

![image](https://github.com/user-attachments/assets/76b274be-5bbf-4943-9f82-d8d9668072be)

Can someone help me?

**Comment by mpkorstanje** [comment]
This sounds like a bug report. You can create one at https://github.com/cucumber/vscode and I suspect the solution will be similar to https://github.com/cucumber/language-service/pull/202.

Do keep in mind that this project is mostly run by volunteers and in peoples spare time. Maintaining an IDE extension is especially time consuming. So it is likely that you'll have to investigate and debug this yourself. For example you could perhaps check if this is a problem specifically caused by using the Spanish annotations instead of the English annotations. That should help narrow things down for people.

**Comment by kieran-ryan** [comment, marked as answer]
Cheers @Tavo521, this is due to a present gap for supporting localised annotations in the underlying Language Service (see [language-service#117](https://github.com/cucumber/language-service/issues/117)) i.e. only detects the default English dialect for step definitions at present.

There is a PR open on this ([language-service#153](https://github.com/cucumber/language-service/pull/153)), but it requires scoping (through configuration) to the appropriate annotations only; rather than searching for all possible localisations through tree-sitter - which would have a performance impact.

Would definitely encourage to upvote the linked issue; but if this is an implementation you think you may able to help with (even partially) or have views on a suitable implementation, feel free to open a PR or leave a comment on one of those issues and I would be happy to support you.
